src/Text/LLVM/DebugUtils.hs

```diff
@@ -447,3 +439,75 @@ guessTypeInfo mdMap name =
           = Just (debugInfoToInfo mdMap di)
 
     go _ = Nothing
+
+------------------------------------------------------------------------
+
+-- | Find source-level names of function arguments
+debugInfoArgNames :: Module -> Define -> IntMap String
```

Comment:
After thinking more about this, I'm not so sure. `Ident` is used for:

* [names of types](https://github.com/elliottt/llvm-pretty/blob/d2fa59fc2e2f32df3d720ed3bde41924a0ec558e/src/Text/LLVM/AST.hs#L390), 
* [names of function arguments](https://github.com/elliottt/llvm-pretty/blob/d2fa59fc2e2f32df3d720ed3bde41924a0ec558e/src/Text/LLVM/AST.hs#L448), 
* [names of blocks](https://github.com/elliottt/llvm-pretty/blob/d2fa59fc2e2f32df3d720ed3bde41924a0ec558e/src/Text/LLVM/AST.hs#L502), and 
* [names of variables (instruction results)](https://github.com/elliottt/llvm-pretty/blob/d2fa59fc2e2f32df3d720ed3bde41924a0ec558e/src/Text/LLVM/AST.hs#L1065).

For `debugInfoArgNames`, the values are source-level identifiers, so they have no correspondence with LLVM identifiers. For `debugInfoGlobalLines` and `debugInfoDefineLines`, it might make more sense to use a `Symbol`. However, it's also not clear to me that [`DIGlobalVariable`](https://llvm.org/docs/LangRef.html#diglobalvariable) and [`DISubprogram`](https://llvm.org/docs/LangRef.html#disubprogram) are guaranteed to have names that actually correspond to the definition's name in the LLVM module, so perhaps it would be best to avoid `Symbol` as well to avoid insinuating that they do. Thoughts @RyanGlScott?

Reply:
You've convinced me. In that case, I'm fine with sticking with the types as-is.